ctapipe/image/extractor.py

```diff
@@ -1006,22 +1003,16 @@ def _apply_second_pass(
         else:
             mask_biggest = mask_clean
 
+        # for all pixels except the core ones in the main island of the
+        # preliminary image, the waveform will be integrated once more (2nd pass)
+
+        mask_2nd_pass = ~mask_biggest | (mask_biggest & (charge_1stpass < core_th))
```

Comment:
Isn't this easier done with
`mask_2nd_pass = ~ (mask_biggest & (charge_1stpass > core_th))`
?

Also I think this can be done after the part where you check that the image has at least 3 pixels

Reply:
> Isn't this easier done with
> `mask_2nd_pass = ~ (mask_biggest & (charge_1stpass > core_th))`?

I wrote in the way that for me was more readable, and I still think it is

> Also I think this can be done after the part where you check that the image has at least 3 pixels

It shouldn't change the result, but yes it makes more sense logically